api/v1alpha1/storageconsumer_types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2021 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageConsumerState represent a StorageConsumer's state
+type StorageConsumerState string
+
+const (
+	// StateReady represents Ready state of an object
+	StateReady       StorageConsumerState = "Ready"
+	// StateConfiguring represents Configuring state of an object
+	StateConfiguring StorageConsumerState = "Configuring"
+	// StateDeleting represents Deleting state of an object
+	StateDeleting    StorageConsumerState = "Deleting"
+	// StateFailed represents Failed state of an object
+	StateFailed      StorageConsumerState = "Failed"
+)
+
+// StorageConsumerSpec defines the desired state of StorageConsumer
+type StorageConsumerSpec struct {
+	// Capacity is the total quota size allocated to a consumer.
+	// +optional
```

Comment:
Why is the capacity field optional?

Reply:
Capacity is optional because as we don't want to create the underlying rook resources prior to the connection request. Based on the earlier discussion, the rook resources will be created on-demand. (ie) only when capacity is defined, the rook resource creation will be proceeded.

Reply:
@kesavanvt Do we need to revise this decision now that we decided to move to a ticket-based onboarding?

Reply:
Yes, with reference to ticker-based onboarding, its expected that the StorageConsumer is created with Capacity by the API server. Have Removed optional .